users/api/logging.go

```diff
@@ -30,12 +30,19 @@ func LoggingMiddleware(svc users.Service, logger mglog.Logger) users.Service {
 // If the request fails, it logs the error.
 func (lm *loggingMiddleware) RegisterClient(ctx context.Context, token string, client mgclients.Client) (c mgclients.Client, err error) {
 	defer func(begin time.Time) {
-		message := fmt.Sprintf("Method register_client with id %s using token %s took %s to complete", c.ID, token, time.Since(begin))
+		fields := map[string]interface{}{
+			"method":   "register_client",
+			"id":       c.ID,
+			"token":    token,
+			"duration": time.Since(begin),
+		}
+		message := fmt.Sprintf("Method %s with id %s using token %s took %s to complete", fields["method"], fields["id"], fields["token"], fields["duration"])
 		if err != nil {
-			lm.logger.Warn(fmt.Sprintf("%s with error: %s.", message, err))
+			fields["error"] = err.Error()
+			lm.logger.Warn(ctx, fmt.Sprintf("%s with error: %s.", message, fields["error"]))
 			return
 		}
-		lm.logger.Info(fmt.Sprintf("%s without errors.", message))
+		lm.logger.Info(ctx, fmt.Sprintf("%s without errors.", message))
 	}(time.Now())
```

Comment:
Something like below give output as JSON, which can be used for filtering
```suggestion
		fields := map[string]interface{}{
			"method":   "register_client",
			"id":       c.ID,
			"token":    token,
			"duration": time.Since(begin),
            "error" : "no errors / without errors"
		}
		if err != nil {
			fields["error"] = err.Error()
		}
		lm.logger.Info(ctx, fields)
	}(time.Now())
```
But At Present , it retruns as plain sentence text, How we will apply filtering on the plain text?